### PR TITLE
Support Minitest spec expectations

### PR DESCRIPTION
**Pull request by faragorn**
WIP

Any feedback would be very much appreciated.

Will add some test cases as well

### CODE REVIEW

**Comment by alto** [comment]
@faragorn Your work / pull request is highly appreciated. I'm 100% backing helpers for minitest. I'm using minitest myself in all projects, and _AASM_ is the only one with rspec.

Thanks a lot for your help!

**Comment by faragorn** [comment]
@alto pretty much ready for review :)

**Comment by alto** [comment]
Hi Faragorn, thanks a lot for this PR. I really appreciate the effort you put into this!

I have a question. What are the files `test/minitest_helper.rb` and `test/unit/minitest_matcher_test.rb` for? It seems like you are trying to test your minitest helpers, but – unfortunately – these files are not run at all during `rake` (used by _Travis CI_). Maybe you were heading for running `rake test` at some point (and forgot to add to `travis.yml`?).

Anyhow, running `rake test` works fine as expected (when called manually).

Another question: I have never seen `infect_an_assertion` before. What does it do? (I have to admit I couldn't find anything of use in the net about it).

**Comment by alto** [comment]
Oh, and could you add something to the README, please? I wanted to start playing around with your code but couldn't find anything to start with (before reading the code). Bummer :)

**Comment by faragorn** [comment]
@alto 
> What are the files test/minitest_helper.rb and test/unit/minitest_matcher_test.rb for? It seems like you are trying to test your minitest helpers, but – unfortunately – these files are not run at all during rake (used by Travis CI). Maybe you were heading for running rake test at some point (and forgot to add to travis.yml?).
Exactly I am trying to test my minitest helpers with minitest setting. Yes I realized it later, gonna fix travis.yml

> Another question: I have never seen `infect_an_assertion` before. What does it do? (I have to admit I couldn't find anything of use in the net about it).
`infect_an_assertion` simply calls an assertion method and passes arguments to it

I will update a README as well.

**Comment by faragorn** [comment]
@alto I think now it is ready

File: README.md [hunk 1]
```diff
@@ -1005,6 +1007,90 @@ expect(multiple).to allow_transition_to(:processing).on(:move)
 expect(multiple).to_not allow_transition_to(:sleeping).on(:move)
 ```
 
+#### Minitest
+
+AASM provides assertions and rspec-like expectations for [Minitest](https://github.com/blowmage/minitest-rails).
```
Comment (alto) [inline README.md]: I think `minitest` is provided by [seattlerb](https://github.com/seattlerb/minitest) (and Ryan would be really sad if we prefer the Rails wrapper gem).

**Comment by alto** [comment]
@faragorn Your PR is almost perfect. Please have a look at my last comment (regarding `seattlerb`). 

Thanks a lot for your work!

File: README.md (hunk 1 above)
Comment (faragorn) [inline README.md]: no worries, that's easy to fix, lets not disappoint Ryan

**Comment by faragorn** [comment]
@alto addressed your last comment

**Comment by taschetto** [comment]
+1 for this. :)

**Comment by joaosoares** [comment]
Is there anything left to do to get this merged that I can help with? (Rebasing, additional modifications, etc.)

**Comment by anilmaurya** [comment]
@joaosoares Thank you for helping hand, I will review and merge it soon if @alto is not available to merge.

**Comment by joaosoares** [comment]
Apparently the reason the builds are failing is that the url for dynamo db in `.circle.yml` now has an extra `.tar.gz` at the end (as per the newer .circle.yml file in master), so wget finishes with an error.

**Comment by anilmaurya** [comment]
Thank you @faragorn 🏆

@joaosoares Thank you for follow-up. 😊

**Comment by faragorn** [comment]
Awesome, thanks!
@anilmaurya Please feel free to tag me if there are any issues with this.

**Comment by alto** [comment]
@faragorn Sorry for getting back to this so late. Thanks a lot for your awesome work!

I'm happy @anilmaurya merged this PR in the meantime.

**Comment by fightingtheboss** [comment]
@alto @anilmaurya Would be great to cut a version with the minitest expectations included! For now I'm relying on the specific commit...

**Comment by anilmaurya** [comment]
@fightingtheboss Thank you for reminding , I will release `4.12.1` soon.

**Pull request merged**